lib/probes/http.js

```diff
@@ -523,24 +526,26 @@ function patchServer (module, protocol) {
         // it replies with the txname that was actually used if there was an error
         const txname = ao.reporter.sendHttpSpan(args);
 
-        // they should match (except for the domain prefix)
-        if (txname && args.txname && txname !== args.txname
-          && ao.cfg.domainPrefix
-          && txname.indexOf(args.txname) + args.txname.length !== txname.length) {
-          log.warn(
-            'sendHttpSpan() changed TransactionName from %s to %s',
-            args.txname,
-            txname
-          )
-        }
+        // TODO BAM the txname not null check can be removed once appoptics-bindings has been
+        // changed to return the integer error code instead of a null string.
+        if (typeof txname === 'string' && txname) {
+          // if there is a txname and it doesn't match
+          if (args.txname && txname !== args.txname) {
+            // here the names don't match so we might need to log a warning.
+            // if there isn't a domain prefix warn about the difference and if there
+            // is a domain prefix warn if the trailing txname doesn't match.
+            if (!ao.cfg.domainPrefix || !txname.endsWith(txname)) {
+              log.warn(`sendHttpSpan() changed TransactionName from ${args.txname} to ${txname}`);
+            }
+          }
 
-        // what to do if an error sending? don't know transaction name but shouldn't
-        // matter, so don't use the return value.
-        if (txname) {
-          exitKeyValuePairs.TransactionName = txname
+          // if it's a string the worst it can be is an empty string.
+          exitKeyValuePairs.TransactionName = txname;
         } else {
-          log.error('sendHttpSpan() returned empty TransactionName')
+          exitKeyValuePairs.TransactionName = args.txname || 'unknown';
+          dbSendError.log(`sendHttpSpan() code ${txname}`);
```

Comment:
out of curiosity: When are these Debounced logs sent?

Reply:
the default debouncing window is 1 second or 100 instances of the error. they're settable at construction time. right now i should fix the Debounce code to reset both `lastTime` and `lastCount` when a log is emitted; currently they're independent.